testing/functional_tests/test_cascaded_architecture_tutorial.py

```diff
@@ -0,0 +1,22 @@
+import logging
+import os
+import pytest
+from pytest_console_scripts import script_runner
+from testing.functional_tests.t_utils import __data_testing_dir__
+from testing.common_testing_util import download_dataset
+
+logger = logging.getLogger(__name__)
+
+@pytest.mark.script_launch_mode('subprocess')
+def test_cascaded_architecture_tutorial(script_runner):
+    download_dataset("data_example_spinegeneric")
+    file_config = os.path.join(__data_testing_dir__, 'cascaded_architecture_tutorial_config.json')
+
+    ret = script_runner.run('ivadomed', '-c', f'{file_config}',
```

Comment:
because of this function which gets executed in ivadomed main
`
def __get_commit(path_to_git_folder=None):
    """Get GIT ivadomed commit.

    Args:
        path_to_git_folder (str): Path to GIT folder.
    Returns:
        str: git commit ID, with trailing '*' if modified.
    """
    if path_to_git_folder is None:
        path_to_git_folder = __ivadomed_dir__
    else:
        path_to_git_folder = os.path.abspath(os.path.expanduser(path_to_git_folder))

    p = subprocess.Popen(["git", "rev-parse", "HEAD"], stdout=subprocess.PIPE, stderr=subprocess.PIPE,
                         cwd=path_to_git_folder)
`
since the existance of os.path.abspath it assumes that current working directory is ivadomed root (which is not unless you run pytest inside the root directory) we can only execute this test inside ivadomed root directory or else it would flail

Reply:
> since the existance of os.path.abspath it assumes that current working directory is ivadomed root (which is not unless you run pytest inside the root directory) we can only execute this test inside ivadomed root directory or else it would flail

good catch @cakester ! it would be great to solve that issue.

Reply:
also, places such as
```
    def write_derivatives_dataset_description(self, path_data):
        """Writes default dataset_description.json file if not found in path_data/derivatives folder
        """
        filename = 'dataset_description'
        deriv_desc_file = '{}/derivatives/{}.json'.format(path_data, filename)
        label_desc_file = '{}/derivatives/labels/{}.json'.format(path_data, filename)
        # need to write default dataset_description.json file if not found
        if not os.path.isfile(deriv_desc_file) and not os.path.isfile(label_desc_file):
            f = open(deriv_desc_file, 'w')
            f.write('{"Name": "Example dataset", "BIDSVersion": "1.0.2", "PipelineDescription": {"Name": "Example pipeline"}}')
            f.close()
```
assumes that path data is relative to the current running directory, which is not when we execute tests in the testing directory, that further prevents this test to be run inside testing dir

Reply:
problem solved, we change directory before running the tests!

Reply:
> problem solved, we change directory before running the tests!

is this directory change now implemented? is it done automatically in the code?

Reply:
it's only done with this test, do you want it for all of ivadomed?